modules/Microsoft.RecoveryServices/vaults/deploy.bicep

```diff
@@ -112,11 +112,10 @@ param securitySettings object = {}
 
 @description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set.')
 @allowed([
-  ''
   'Enabled'
   'Disabled'
 ])
-param publicNetworkAccess string = ''
+param publicNetworkAccess string = 'Disabled'
```

Comment:
Up until now we used `null` as a default value if nothing was provided (including no Private Endpoints). So I take it that `null` isn't a valid value anymore?

If that's the case, I think we should also simplify the logic `publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(privateEndpoints) ? 'Disabled' : null)` itself, as the parameter cannot be `empty` anymore.

I think so far we used `''` as a default as we figured a user wouldn't need to provide the value - and only if he/she did we'd use it. Otherwise we'd assume a value based on whether a private endpoint was provided or not.

If we change this because it isn't working anymore, also the parameter's description must be updated.

Reply:
Changes committed as addressed in the review. Changing to assign param instead of logic due to the unsupported nature of `null` for the mapped NetworkSettings param during deployment